utils.go

```diff
@@ -22,18 +23,11 @@ func Factorial(n int) *big.Int {
 
 //  Returns 2 primes such that p = 2 * q + 1 and that the length of
 //  p is nbits.  `p` is called a safe prime
+//
+// Deprecated: This function has been left here just for backward compatibility.
+// Please use `GenerateSafePrime` from the `safe_prime_generator.go` directly.
 func GenerateSafePrimes(nbits int, random io.Reader) (p, q *big.Int, err error) {
-	for {
-		q, err = rand.Prime(random, nbits-1)
-		if err != nil {
-			return
-		}
-		p = (new(big.Int)).Mul(q, big.NewInt(2))
-		p = p.Add(p, big.NewInt(1))
-		if p.ProbablyPrime(50) { //a probability of 2**-100 of not being prime
-			return
-		}
-	}
+	return GenerateSafePrime(nbits, 4, 120*time.Second)
```

Comment:
I'm not sure about this `GenerateSafePrimes` function to call `GenerateSafePrime`, it's little bit duplication. Do we really need to keep this `GenerateSafePrimes` function instead of switching our tecdsa implementation to use the new function?

Reply:
Yeah, good point. It's not JVM with people locked on shapshot, we are referring to a specific version anyway. No need to be such super backward-compatible. Especially that we want to drop a support for 1.3 / 1.4.